xonsh/proc.py

```diff
@@ -528,3 +528,33 @@ def _wcode_to_popen(code):
     else:
         # Can this happen? Let's find out. Returning None is not an option.
         raise ValueError("Invalid os.wait code: {}".format(code))
+
+
+_CCTuple = namedtuple("_CCTuple", ["stdin",
+                                   "stdout",
+                                   "stderr",
+                                   "pid",
+                                   "returncode",
+                                   "args",
+                                   "alias",
+                                   "stdin_redirect",
+                                   "stdout_redirect",
+                                   "stderr_redirect"])
+
+class CompletedCommand(_CCTuple):
+    """
+    Represents a completed subprocess-mode command.
+    """
+
+    def __bool__(self):
+        return self.returncode == 0
+
+    def __eq__(self, other):
+        return self.returncode == other
+
+def make_completed_command(**kwargs):
```

Comment:
That was my initial thought, too (perhaps naturally), but `namedtuple` does some weird stuff with `__new__`, and this was the path of least resistance.  I'm happy to look into it a bit more, but this seemed to work with much less pain than I imagine the alternative would take.